viewer/packages/tools/src/Toolbar/ToolbarTool.ts

```diff
@@ -0,0 +1,181 @@
+  private readonly _handleScreenshotListener = this.saveScreenShot.bind(this);
+  private readonly _handleChangeCameraTargetListener = this.changeCameraTargetOnClick.bind(this);
+  private readonly _handleCameraZoomPastCursorListener = this.toggleCameraZoomPastToCursor.bind(this);
+  private readonly _handleFitCameraToModelListener = this.fitCameraToAllModels.bind(this);
+
+  constructor(viewer: Cognite3DViewer) {
+    super();
+
+    this._viewer = viewer;
+    this._toolbar = new Toolbar(this._viewer);
+
+    const cameraManager = this._viewer.cameraManager as DefaultCameraManager;
+    const cameraControlOption = cameraManager.getCameraControlsOptions();
+    this._cameraOptions[0] = cameraControlOption.changeCameraTargetOnClick;
+    if (cameraControlOption.mouseWheelAction === 'zoomPastCursor') {
+      this._cameraOptions[1] = false;
+    } else {
+      this._cameraOptions[1] = true;
+    }
+  }
+
+  /**
+   * Add a Icon button into the Toolbar container
+   * @param backgroundImageUri Background image to be placed onto icon
+   * @param onClick Click event callback function which will be used to perform custom functionlity of the user
+   * @param toolTip Optional tooltip for the icon button
+   */
+  public addToolbarButton(backgroundImageUri: string, onClick: () => void, toolTip: string = ''): void {
+    this._toolbar.addToolbarButton(backgroundImageUri, false, onClick, false, toolTip);
+  }
+
+  /**
+   * Add a Icon toggle button into the Toolbar container
+   * @param backgroundImageUri Background image to be placed onto icon
+   * @param onToggled Click event callback function which will be used to perform custom functionlity of the user
+   * @param isActive Is the feature active by default.
+   * @param toolTip Optional tooltip for the icon button
+   */
+  public addToolbarToggleButton(
+    backgroundImageUri: string,
+    onToggled: () => void,
+    isActive: boolean,
+    toolTip: string = ''
+  ): void {
+    this._toolbar.addToolbarButton(backgroundImageUri, true, onToggled, isActive, toolTip);
+  }
+
+  /**
+   * Set the position of the toolbar container
+   * @param position ToolbarPosition value such as Top, Bottom, Left, Right within the canvas
+   */
+  public setPosition(position: ToolbarPosition): void {
+    this._toolbar.setPosition(position);
+  }
+
+  /**
+   * Add Enable or Disable Axis view tool to the Toolbar
+   */
+  public addAxisToolToggle(): void {
+    this.addToolbarToggleButton(svgAxisIcon, this._handleAxisViewToolListener, false, 'Axis Tool');
+  }
+
+  /**
+   * Add Save the screenshot of the canvas been rendered to the Toolbar
+   */
+  public addTakeScreenshotTool(): void {
+    this.addToolbarButton(svgScreenshotIcon, this._handleScreenshotListener, 'Take Screenshot');
+  }
+
+  /**
+   * Adds Toggle enabling/disabling the Camera target on mouse click to the Toolbar
+   */
+  public addCameraTargetOnClickToggle(): void {
```

Comment:
This function assumes that change target initally. Use state from camera manager.